### PR TITLE
Cleanup HHctl interop

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Cleanup HHctl interop

I have verified all the code still works

E.g. `ShowPopup`
Before:
Before - Popup

After:
After - Popup

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/3173)

### CODE REVIEW

File: src/System.Windows.Forms.Primitives/src/Interop/Interop.Libraries.cs [hunk 1]
```diff
@@ -9,6 +9,7 @@ public static class Libraries
         public const string Comctl32 = "comctl32.dll";
         public const string Comdlg32 = "comdlg32.dll";
         public const string Gdi32 = "gdi32.dll";
+        public const string Hhctrl = "hhctrl.ocx";
```
Comment (gpetrou) [inline src/System.Windows.Forms.Primitives/src/Interop/Interop.Libraries.cs:12]: Should this be removed from ExternDll.cs now?

File: src/System.Windows.Forms.Primitives/src/Interop/Hhctrl/Interop.HtmlHelpW.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal static partial class Hhctl
+    {
+        [DllImport(Libraries.Hhctrl, CharSet = CharSet.Unicode, ExactSpelling = true)]
+        public static extern int HtmlHelpW(IntPtr hwndCaller, string pszFile, HH uCommand, IntPtr dwData);
```
Comment (gpetrou) [inline src/System.Windows.Forms.Primitives/src/Interop/Hhctrl/Interop.HtmlHelpW.cs:13]: Do these need to be updated for the nullable changes?

Comment (hughbe) [reply src/System.Windows.Forms.Primitives/src/Interop/Hhctrl/Interop.HtmlHelpW.cs:13]: Will rebase

File: src/System.Windows.Forms.Primitives/src/Interop/Interop.Libraries.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Windows.Forms.Primitives/src/Interop/Interop.Libraries.cs:12]: Yes sir

File: src/System.Windows.Forms.Primitives/src/Interop/Hhctrl/Interop.HH_POPUPW.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+internal partial class Interop
+{
+    internal static partial class Hhctl
+    {
+        public unsafe struct HH_POPUPW
+        {
+            public int cbStruct;
+            public IntPtr hinst;
+            public int idString;
```
Comment (RussKie) [inline src/System.Windows.Forms.Primitives/src/Interop/Hhctrl/Interop.HH_POPUPW.cs:16]: The doc says it's a `uint`
https://docs.microsoft.com/en-us/windows/win32/api/htmlhelp/ns-htmlhelp-hh_popup

File: src/System.Windows.Forms.Primitives/src/Interop/Kernel32/Interop.GetShortPathNameW.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Text;
+
+internal static partial class Interop
+{
+    internal static partial class Kernel32
+    {
+        [DllImport(Libraries.Kernel32, CharSet = CharSet.Unicode, ExactSpelling = true)]
```
Comment (RussKie) [inline src/System.Windows.Forms.Primitives/src/Interop/Kernel32/Interop.GetShortPathNameW.cs:12]: Missing `SetLastError`
https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getshortpathnamew#return-value

File: src/System.Windows.Forms/src/System/Windows/Forms/Help.cs [hunk 5]
```diff
@@ -99,47 +92,40 @@ public static void ShowHelp(Control parent, string url, HelpNavigator command, o
...
 
-            try
+            var pop = new HH_POPUPW
+            {
+                cbStruct = sizeof(HH_POPUPW),
+                idString = 0,
+                pt = location,
+                rcMargins = new RECT(-1, -1, -1, -1), // Ignore
+                clrForeground = -1, // Ignore
+                clrBackground = Color.FromKnownColor(KnownColor.Window).ToArgb() & 0x00ffffff
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Help.cs:114]: Is this equivalent to calling `COLORREF.ColorToCOLORREF(KnownColor.Window)`?

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/Help.cs:114]: Sort of: more like `COLORREF.ColorToCOLORREF(SystemColors.Window)`

**Pull request merged**